### ISSUE TITLE
Releases >3.6.1 (not inclusive) newrelic_entity_tags resource throws an error

### ISSUE THREAD

**Issue by travisroberts-sf** [newrelic/terraform-provider-newrelic#2115]
When creating new resources (in my case nrql alert conditions and synthetic transaction), the newrelic_entity_tags resource is defined as follows:
```hcl
resource "newrelic_entity_tags" "linux_file_count_alert_condition_tags" {
  guid = newrelic_nrql_alert_condition.linux_file_count_alert_condition.entity_guid

  dynamic "tag" {
    for_each = var.tags
    content {
      key    = tag.key
      values = list(tag.value)
    }
  }

  depends_on = [newrelic_nrql_alert_condition.linux_file_count_alert_condition]
}
```
Prior to release 3.7.0 this worked without error.  Since release 3.7.0 however, we are getting an error:
```Error: Missing required argument

  on ../../modules/alerting/alert_conditions/infrastructure/file_count/nr_condition_linux_file_count.tf line 78, in resource "newrelic_entity_tags" "linux_file_count_alert_condition_tags":
  78:   guid = newrelic_nrql_alert_condition.linux_file_count_alert_condition.entity_guid

The argument "guid" is required, but no definition was found.
```
We are able to get around this error by running the apply a second time, but in our production environment that is not ideal as terraform is run from a pipeline.  

### Terraform Version
Terraform v0.13.7

### Affected Resource(s)
As far as I can tell it's any resource having tags applied to it with 'newrelic_entity_tags'

### Terraform Configuration
The following is the provider configuration where I am seeing this bug, but it also shows same behavior in 3.7.1
```hcl
terraform {
  required_providers {
    newrelic = {
      source  = "newrelic/newrelic"
      version = "= 3.7.0" 
    }
  }
  required_version = ">= 0.13"
}
```

### Actual Behavior
When creating an alert condition (or synthetic transaction), I am getting the following error and no tags are applied to the resource:
```
Error: Missing required argument

  on ../../modules/alerting/alert_conditions/infrastructure/file_count/nr_condition_linux_file_count.tf line 78, in resource "newrelic_entity_tags" "linux_file_count_alert_condition_tags":
  78:   guid = newrelic_nrql_alert_condition.linux_file_count_alert_condition.entity_guid

The argument "guid" is required, but no definition was found.
```
If I run the terraform apply a second time, the error goes away and tags are applied.

### Expected Behavior
When using anything older than, and including, Release 3.6.1, this issue does not occur and all tags are applied correctly the first time.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Set the provider version to 3.7.0 (or 3.7.1) and apply to any new resource with tags
2. `terraform apply`
3. `terraform apply` a second time and tags are applied

1. Set the provider version to 3.6.1 (or older) and apply to new resource with tags
2. `terraform apply`

### Important Factoids
There is nothing special about the code I am applying however it is worth mentioning that this same issue used to occur in much older version of the provider (<=2.50.2) and looked to have been resolved since version 3.0.0.

**Comment by utr1903** [comment]
@travisroberts-sf +1 from my side. Was about the open an issue myself :)

**Comment by darrensmithwtc** [comment]
I'm also facing this issue 👍

**Comment by NSSPKrishna** [comment]
Released a possible fix in version v3.10.0. I'm going to close this for now. Please reply if you experience the error again and I will reopen.
